platforms/ios/lib/WysiwygComposer/Sources/HTMLParser/Extensions/DTCoreText/DTHTMLElement.swift

```diff
@@ -35,4 +35,70 @@ extension DTHTMLElement {
             }
         }
     }
+
+    func clearTrailingAndLeadingNewlinesInCodeblocks() {
+        guard let childNodes = childNodes as? [DTHTMLElement] else {
+            return
+        }
+
+        if name == "pre",
+           childNodes.count == 1,
+           let child = childNodes.first as? DTTextHTMLElement,
+           var text = child.text(),
+           text != .nbsp {
+            var leadingDiscardableElement: DiscardableTextHTMLElement?
+            var trailingDiscardableElement: DiscardableTextHTMLElement?
+            var shouldReplaceNodes = false
+
+            if text.hasPrefix("\(Character.nbsp)") {
+                shouldReplaceNodes = true
+                text.removeFirst()
+                leadingDiscardableElement = createDiscardableElement()
+            }
+            
+            if text.hasSuffix("\(Character.nbsp)") {
+                shouldReplaceNodes = true
+                text.removeLast()
+                trailingDiscardableElement = createDiscardableElement()
+            }
+
+            if shouldReplaceNodes {
```

Comment:
I'm guessing this could be replaced by 

```swift
let hasLeadingNbsp = text.hasPrefix(...)
let hasTrailingNbsp = text.hasSuffix(...)

if hasLeadingNbsp || hasTrailingNbsp {
    removeAllChildNodes() 
    
    // Prepare nodes and push them, we only need to create the new nodes at this point
    // Don't need to work with these optionals.
    [...] 
}
```